client/client_test.go

```diff
@@ -59,7 +59,6 @@ func (ss *notifyingSender) reset(waiter *sync.WaitGroup) {
 
 func (ss *notifyingSender) wait() {
 	ss.waiter.Wait()
```

Comment:
`ss.waiter` may already be nil here, which will panic. We need to keep track of whether we have notified with a separate flag instead of using the shared `ss.waiter` field.